tests/test_spatial_analysis/test_spatial_interaction.py

```diff
@@ -285,6 +360,56 @@ def test_new_ax_has_right_titles(self):
                     error_msg
                 )
 
+    def test_sinlge_stratify_by(self):
+        ax_dict = spatial_interaction(
+            self.adata,
+            "cluster_num",
+            "Neighborhood Enrichment",
+            stratify_by="cluster_str"
+        )
+        unique_cluster_str_values = self.adata.obs["cluster_str"].unique()
+
+        # Get the keys (unique cluster values) from the dictionary
+        keys = list(ax_dict.keys())
+
+        # Assert that we have at least two keys (clusters)
+        self.assertEqual(len(keys), 2)
+
+        # Assert that the axes associated
+        # with the first and second keys are different
+        self.assertNotEqual(ax_dict[keys[0]], ax_dict[keys[1]])

```

Comment:
@ruiheesi, we know the actual values of the keys. Would you please change the assert to check that the list in ax_dict matches the ground truth we know?

Reply:
Hi George, this portion of the test is aiming at testing general behavior, the ground truth is tested in later tests